### ISSUE TITLE
h5的swiper组件绑定了current，当current为0时swiper不切换，其他值可以正常切换

### ISSUE THREAD

**Issue by gdpu-wxq** [NervJS/taro#4492]
**问题描述**
h5环境，swiper组件绑定了current，当current为0时swiper不切换，其他值可以正常切换

**复现步骤**
[复现问题的步骤]
H5环境，swiper组件绑定了current，动态修改current，当current=0的时候，swiper不切换

**期望行为**
希望当current为0的时候，切换到相应的swiper

我目前使用的是1.3.18版本，还未解决。
这个是之前的github提出来的
[问题的原始链接](https://github.com/NervJS/taro/issues/4052#issue-476066571)

**Comment by jinjinjin0731** [comment]
升级到最新试下

**Comment by gdpu-wxq** [comment]
> 升级到最新试下
还是不行

**Comment by gdpu-wxq** [comment]
还是不行  @jinjinjin0731

**Comment by jinjinjin0731** [comment]
@gdpu-wxq  贴下代码。  没有复现

**Comment by gdpu-wxq** [comment]
@jinjinjin0731 
`import Taro from '@tarojs/taro'
import { Swiper, View, SwiperItem } from '@tarojs/components'
import './index.scss'
interface IState {
    current: number
 }

/**
 * 购物车页面
 */
export default class extends Taro.Component<{}, IState> {
    config: Taro.PageConfig = {
        navigationBarTitleText: '...',
    }

    constructor() {
        super(...arguments)

        this.state = {
            current: 0
        }
    }

    async componentDidShow() {

    }

    goToBefore (current) {
        this.setState({
            current
        })
    }
    render() {
        const {current } = this.state
        return (
            <View>
                <Swiper
                className='test-h'
                indicatorColor='#999'
                indicatorActiveColor='#333'
                vertical
                circular
                indicatorDots
                current={current}
                onChange={(e)=>{this.setState({current:e.detail.current})}}>
                <SwiperItem>
                    <View className='demo-text-1'>1</View>
                </SwiperItem>
                <SwiperItem>
                    <View className='demo-text-2'>2</View>
                </SwiperItem>
                <SwiperItem>
                    <View className='demo-text-3'>3</View>
                </SwiperItem>
            </Swiper>
                <View onClick={ this.goToBefore.bind(this, 0) }>0</View>
                <View onClick={ this.goToBefore.bind(this, 1) }>1</View>
                <View onClick={ this.goToBefore.bind(this, 2) }>2</View>
            
            </View>
            
        )
    }
}
`

**Comment by gdpu-wxq** [comment]
`
import Taro from '@tarojs/taro'
import { Swiper, View, SwiperItem } from '@tarojs/components'
import './index.scss'
interface IState {
    current: number
 }

/**
 * 购物车页面
 */
export default class extends Taro.Component<{}, IState> {
    config: Taro.PageConfig = {
        navigationBarTitleText: '...',
    }

    constructor() {
        super(...arguments)

        this.state = {
            current: 0
        }
    }

    async componentDidShow() {

    }

    goToBefore (current) {
        this.setState({
            current
        })
    }
    render() {
        const {current } = this.state
        return (
            <View>
                <Swiper
                className='test-h'
                indicatorColor='#999'
                indicatorActiveColor='#333'
                vertical
                circular
                indicatorDots
                current={current}
                onChange={(e)=>{this.setState({current:e.detail.current})}}>
                <SwiperItem>
                    <View className='demo-text-1'>1</View>
                </SwiperItem>
                <SwiperItem>
                    <View className='demo-text-2'>2</View>
                </SwiperItem>
                <SwiperItem>
                    <View className='demo-text-3'>3</View>
                </SwiperItem>
            </Swiper>
                <View onClick={ this.goToBefore.bind(this, 0) }>0</View>
                <View onClick={ this.goToBefore.bind(this, 1) }>1</View>
                <View onClick={ this.goToBefore.bind(this, 2) }>2</View>
            
            </View>
            
        )
    }
}

`

**Comment by gdpu-wxq** [comment]
`import Taro from '@tarojs/taro'
import { Swiper, View, SwiperItem } from '@tarojs/components'
import './index.scss'
interface IState {
    current: number
 }

/**
 * 购物车页面
 */
export default class extends Taro.Component<{}, IState> {
    config: Taro.PageConfig = {
        navigationBarTitleText: '...',
    }

    constructor() {
        super(...arguments)

        this.state = {
            current: 0
        }
    }

    async componentDidShow() {

    }

    goToBefore (current) {
        this.setState({
            current
        })
    }
    render() {
        const {current } = this.state
        return (
            <View>
                <Swiper
                className='test-h'
                indicatorColor='#999'
                indicatorActiveColor='#333'
                vertical
                circular
                indicatorDots
                current={current}
                onChange={(e)=>{this.setState({current:e.detail.current})}}>
                <SwiperItem>
                    <View className='demo-text-1'>1</View>
                </SwiperItem>
                <SwiperItem>
                    <View className='demo-text-2'>2</View>
                </SwiperItem>
                <SwiperItem>
                    <View className='demo-text-3'>3</View>
                </SwiperItem>
            </Swiper>
                <View onClick={ this.goToBefore.bind(this, 0) }>0</View>
                <View onClick={ this.goToBefore.bind(this, 1) }>1</View>
                <View onClick={ this.goToBefore.bind(this, 2) }>2</View>
            
            </View>
            
        )
    }
}
`

**Comment by gdpu-wxq** [comment]
```
import Taro from '@tarojs/taro'
import { Swiper, View, SwiperItem } from '@tarojs/components'
import './index.scss'
interface IState {
    current: number
 }

/**
 * 购物车页面
 */
export default class extends Taro.Component<{}, IState> {
    config: Taro.PageConfig = {
        navigationBarTitleText: '...',
    }

    constructor() {
        super(...arguments)

        this.state = {
            current: 0
        }
    }

    async componentDidShow() {

    }

    goToBefore (current) {
        this.setState({
            current
        })
    }
    render() {
        const {current } = this.state
        return (
            <View>
                <Swiper
                className='test-h'
                indicatorColor='#999'
                indicatorActiveColor='#333'
                vertical
                circular
                indicatorDots
                current={current}
                onChange={(e)=>{this.setState({current:e.detail.current})}}>
                <SwiperItem>
                    <View className='demo-text-1'>1</View>
                </SwiperItem>
                <SwiperItem>
                    <View className='demo-text-2'>2</View>
                </SwiperItem>
                <SwiperItem>
                    <View className='demo-text-3'>3</View>
                </SwiperItem>
            </Swiper>
                <View onClick={ this.goToBefore.bind(this, 0) }>0</View>
                <View onClick={ this.goToBefore.bind(this, 1) }>1</View>
                <View onClick={ this.goToBefore.bind(this, 2) }>2</View>
            
            </View>
            
        )
    }
}
```

**Comment by gdpu-wxq** [comment]
@jinjinjin0731

**Comment by BrightTan** [comment]
我也遇到这个问题了，望解决

**Comment by BrightTan** [comment]
@jinjinjin0731

**Comment by ImPigerla** [comment]
`
// taro version 1.3.4
// 调用方法，index就是跳转的索引
this.refs.testSwiper.mySwiper.slideTo(index)

// swiper ref
<Swiper ref='testSwiper' ...
`

**Comment by lzt9977** [comment]
确实有这个问题

**Comment by sunny2009day** [comment]
多久了,还没解决嘛

**Comment by ZakaryCode** [comment]
fixed in #b4bb27cc5fc5bda53a0ba53379e9e6078eef942f
